lib/demo_mode.rb

```diff
@@ -46,6 +46,16 @@ def current_password=(value)
       Thread.current[:_demo_mode_password] = value
     end
 
+    def session_url(session)
+      routes = DemoMode::Engine.routes
+
+      options = routes.default_url_options.presence
+      options ||= ActionMailer::Base.default_url_options.presence if defined?(ActionMailer::Base)
+      options ||= { only_path: true }
+
+      routes.url_for(controller: 'demo_mode/sessions', action: 'show', id: session, **options)
+    end
```

Comment:
This is a best effort approach to try and generate a URL. We try to resolve the URL in the following order:

1. You can explicitly configure URL generation by setting `DemoMode::Engine.routes.default_url_options`.
2. We'll fallback to the `default_url_options` you've used for `config.action_mailer.default_url_options`. This should work well for the majority of applications and custom configuration probably won't be necessary.
3. We'll fallback to generating just a path.